imsim/stamp.py

```diff
@@ -407,20 +424,15 @@ def draw(self, prof, image, method, offset, config, base, logger):
 
         max_flux_simple = config.get('max_flux_simple', 100)
         faint = self.realized_flux < max_flux_simple
+        bandpass = base['bandpass']
 
-        prof = prof.withFlux(self.realized_flux)
+        prof = prof.withFlux(self.realized_flux, bandpass)
 
         # This seems to be hard-coded to 1 in the imsim code.
         # XXX: Make this a parameter?  Or ok to leave like this?
         # Note: if it's just 1, it would be simpler to just remove it in this function.
         gain = 1.
 
-        if faint or 'sed' not in config:
-            sed = self._trivial_sed
-        else:
-            sed = galsim.config.BuildSED(config, 'sed', base, logger=logger)[0]
-        base['current_sed'] = sed
-
```

Comment:
Did you preserve the essence of this functionality somewhere?  I don't see it.  We want to make sure the faint objects don't bother with all the chromatic stuff.

Reply:
> Did you preserve the essence of this functionality somewhere?

I guess I don't know off hand how to do that.